app/Providers/EloquentServiceProvider.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Providers;
+
+use Illuminate\Support\ServiceProvider;
+use App\Friend;
+
+class EloquentServiceProvider extends ServiceProvider
+{
+    /**
+     * Bootstrap the application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        /**
+         * When deleting a friend, if they are the last record in the database
+         * with the given username, delete the cached avatar from storage.
+         */
+        Friend::deleted(function ($friend) {
```

Comment:
I tend to do these in the models themselves--do you prefer them here for a particular reason? (if so, that's fine, just curious)

I'll usually put it in the model's Boot method:

![image](https://cloud.githubusercontent.com/assets/151829/16087279/710f64a6-32f0-11e6-95de-4e9c56427281.png)